File: app/src/main/java/com/fooxxiie/qrcodemanager/MainActivity.java
package com.fooxxiie.qrcodemanager;

import android.content.DialogInterface;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.widget.EditText;
import android.widget.Toast;


public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

    }

    public void onClickToText(View view) {

        AlertDialog.Builder askTxt = new AlertDialog.Builder(this);

        // Load layout with input
        LayoutInflater inflater = this.getLayoutInflater();
        final View myView = inflater.inflate(R.layout.asktext, null);

        askTxt.setTitle(getResources().getString(R.string.enterTxt))
                .setView(myView)
                .setNegativeButton( getResources().getString(R.string.cancel), null)
                .setPositiveButton( getResources().getString(R.string.btnRun), new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int id) {
                        EditText input = myView.findViewById(R.id.input_text);
                        Intent i = new Intent(MainActivity.this, QrCodeShowText.class);
                        i.putExtra("textQR", input.getText().toString().trim());
                        startActivity(i);
                    }
                })
                .create()
                .show();
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.action_bar, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.action_readQR:
                try {

                    Intent intent = new Intent("com.google.zxing.client.android.SCAN");
                    intent.putExtra("SCAN_MODE", "QR_CODE_MODE"); // "PRODUCT_MODE for bar codes

                    startActivityForResult(intent, 0);

                } catch (Exception e) {

                    Uri marketUri = Uri.parse("market://details?id=com.google.zxing.client.android");
                    Intent marketIntent = new Intent(Intent.ACTION_VIEW,marketUri);
                    startActivity(marketIntent);

                }
                break;
            default:
                break;
        }

        return true;
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (requestCode == 0) {

            if (resultCode == RESULT_OK) {
                String contents = data.getStringExtra("SCAN_RESULT");
                Toast.makeText(this, contents, Toast.LENGTH_SHORT).show();
            }
            if(resultCode == RESULT_CANCELED){
                //handle cancel
            }
        }
    }
}

File: app/src/main/java/com/fooxxiie/qrcodemanager/MainActivity.kt
package com.fooxxiie.qrcodemanager

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.app.AlertDialog
import android.view.MotionEvent
import android.view.View
import android.widget.Toast
import com.google.zxing.integration.android.IntentIntegrator
import com.google.zxing.integration.android.IntentResult
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.form_email.view.*
import kotlinx.android.synthetic.main.form_sms.view.*

class MainActivity : AppCompatActivity() {

    private var typeItems =  arrayOf("Text", "SMS", "Email", "URL", "Contact")

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var ii = IntentIntegrator(this)

        txt_run_scan.setOnClickListener() {
            ii.setBeepEnabled(false)
            ii.initiateScan()
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        var result = IntentIntegrator.parseActivityResult(requestCode, resultCode, data)
        if (result != null) {
            if (result.contents == null) {
                Toast.makeText(applicationContext, "No found data", Toast.LENGTH_SHORT).show()
            } else {
                Toast.makeText(applicationContext, result.contents, Toast.LENGTH_LONG).show()
            }
        } else {
            super.onActivityResult(requestCode, resultCode, data)
        }
    }

    fun onClickActionBtn(view : View) {
        val choices = typeItems
        val builder = AlertDialog.Builder(this)
        with(builder) {
            setTitle(getString(R.string.chooseType))
            setItems(choices) { dialog, which ->
                showAdaptedForm(choices[which])
            }
            setPositiveButton(getString(R.string.cancel), null)
            show()
        }
    }

    private fun showAdaptedForm(choice:String) {
        when (choice) {
            typeItems[0] -> buildAlertForText(typeItems[0])
            typeItems[1] -> buildAlertForSMS(typeItems[1])
            typeItems[2] -> Toast.makeText(applicationContext, getString(R.string.inDev), Toast.LENGTH_SHORT).show()
            typeItems[3] -> Toast.makeText(applicationContext, getString(R.string.inDev), Toast.LENGTH_SHORT).show()
            else -> Toast.makeText(applicationContext, getString(R.string.noData), Toast.LENGTH_SHORT).show()
        }
    }

    private fun buildAlertForText(str: String) {
        var alert = AlertDialog.Builder(this)
        var view = layoutInflater.inflate(R.layout.form_text, null)
        alert.setTitle(str)
                .setView(view)
                .setPositiveButton(getString(R.string.btnRun)) { dialog, p1 ->
                    var i = Intent(applicationContext, QrCodeShowText::class.java)
                    i.putExtra("textQR", view.form_input_text.text.toString())
                    startActivity(i)
                }
                .setNegativeButton(getString(R.string.cancel)) { dialog, p1 ->
                    Toast.makeText(applicationContext, getString(R.string.cancel), Toast.LENGTH_SHORT).show()
                }
                .create()
                .show()
    }

    private fun buildAlertForSMS(str: String) {
        var alert = AlertDialog.Builder(this)
        var view = layoutInflater.inflate(R.layout.form_sms, null)
        alert.setTitle(str)
                .setView(view)
                .setPositiveButton(getString(R.string.btnRun)) { dialog, p1 ->
                    var i = Intent(applicationContext, QrCodeShowText::class.java)
                    var strSms = "sms:" + view.form_input_number.text.toString() + ":" + view.form_input_msg_sms.text.toString()
                    i.putExtra("textQR", strSms)
                    startActivity(i)
                }
                .setNegativeButton(getString(R.string.cancel)) { dialog, p1 ->
                    Toast.makeText(applicationContext, getString(R.string.cancel), Toast.LENGTH_SHORT).show()
                }
                .create()
                .show()
    }
}
